pkg/account/deployer/client.go

```diff
@@ -0,0 +1,268 @@
+	for _, g := range result.GetItems() {
+		if g.GetName() == group.Name {
+			existingGroups = append(existingGroups, g)
+		}
+	}
+
+	if len(existingGroups) == 0 {
+		var createdGroups []accountmanagement.GetGroupDto
+		createdGroups, resp, err = d.client.GroupManagementAPI.CreateGroups(ctx, d.accountInfo.AccountUUID).PutGroupDto([]accountmanagement.PutGroupDto{group}).Execute()
+		if err = d.handleClientResponseError(resp, err, "unable to create group with name: "+group.Name); err != nil {
+			return "", err
+		}
+		if len(createdGroups) < 1 {
+			return "", fmt.Errorf("unable to get UUID of created group with name: %s", group.Name)
+		}
+		return createdGroups[0].GetUuid(), nil
+	}
+
+	if len(existingGroups) > 1 {
+		log.Warn("Updating multiple policies with name %s", group.Name)
+	}
+
+	groupToUpdate := existingGroups[0]
+
+	resp, err = d.client.GroupManagementAPI.EditGroup(ctx, d.accountInfo.AccountUUID, groupToUpdate.GetUuid()).PutGroupDto(group).Execute()
+	if err = d.handleClientResponseError(resp, err, "unable to update group with name: "+group.Name); err != nil {
+		return "", err
+	}
+	return groupToUpdate.GetUuid(), nil
+}
+
+func (d *accountManagementClient) upsertUser(ctx context.Context, userId string) (remoteId, error) {
+	users, resp, err := d.client.UserManagementAPI.GetUsers(ctx, d.accountInfo.AccountUUID).Execute()
+	if err = d.handleClientResponseError(resp, err, "unable to get all users from account "+d.accountInfo.AccountUUID); err != nil {
+		return "", err
+	}
+	for _, u := range users.GetItems() {
+		if u.Email == userId {
+			return userId, nil
+		}
+	}
+
+	resp, err = d.client.UserManagementAPI.CreateUserForAccount(ctx, d.accountInfo.AccountUUID).UserEmailDto(accountmanagement.UserEmailDto{Email: userId}).Execute()
+	if err = d.handleClientResponseError(resp, err, "unable to create user with email: "+userId); err != nil {
+		return "", err
+	}
+
+	return userId, nil
+}
+
+func (d *accountManagementClient) updatePermissions(ctx context.Context, groupId string, permissions []accountmanagement.PermissionsDto) error {
+	if groupId == "" {
+		return fmt.Errorf("group id must not be empty")
+	}
+
+	if len(permissions) == 0 {
+		return nil
+	}
+
+	for _, p := range permissions {
+		if !slices.Contains(d.supportedPermissions, p.PermissionName) {
+			return fmt.Errorf("unsupported permission %q. Must be one of: %v", p.PermissionName, d.supportedPermissions)
+		}
+	}
+	resp, err := d.client.PermissionManagementAPI.AddGroupPermissions(ctx, d.accountInfo.AccountUUID, groupId).PermissionsDto(permissions).Execute()
+	if err = d.handleClientResponseError(resp, err, "unable to update permissions of group with UUID "+groupId); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (d *accountManagementClient) updateAccountPolicyBindings(ctx context.Context, groupId string, policyIds []string) error {
+	if groupId == "" {
+		return fmt.Errorf("group id must not be empty")
+	}
+	if len(policyIds) == 0 {
```

Comment:
same as for `permissions` (and then again for line 230 Env policies and for groups) - is really not a usage error to call these with empty slices?